app/js/referenceSearch.controller.js

```diff
@@ -1,190 +1,193 @@
-        vm.timeoutRefresh;
-
-        //Init method
-        activate();
-
-
-
-        //ng-show logic
-        function isSearching () {
-            return vm.isUserTyping && vm.query.length > 0
-        }
-        function isQueryFieldEmpty () {
-            return vm.query.length == 0 && !vm.isUserTyping;
-        }
-        function conceptsNotFound () {
-            return vm.query.length > 0 && vm.references.length == 0 && !vm.isUserTyping;
-        }
-        function showResults() {
-            return vm.references.length > 0 && vm.query.length > 0 && !vm.isUserTyping;
-        }
-        function showEntriesCount() {
-            return !vm.isUserTyping && vm.references.length && !vm.loadingMorePages;
-        }
-        function showResultInfoAndNavigationButtons () {
-            return !vm.isUserTyping && vm.references.length > 0;
-        }
-        function isNextPagePossible () {
-            return vm.viewRangeEnd() >= vm.references.length;
-        }
-        function isPrevPagePossible() {
-            return vm.pageNumber == 1;
-        }
-
-
-        //Page navigation
-        function nextPage () {
-            vm.pageNumber++;
-        }
-        function prevPage () {
-            vm.pageNumber--;
-        }
-        function firstPage () {
-            vm.pageNumber = 1;
-        }
-        function lastPage () {
-            vm.pageNumber = Math.floor(vm.references.length / vm.entriesPerPage) + 1;
-        }
-
-        //Page dividing
-        function sliceFrom () {
-            return vm.entriesPerPage*vm.pageNumber-vm.entriesPerPage;
-        }
-        function sliceTo () {
-            return vm.entriesPerPage*vm.pageNumber;
-        }
-
-        //Page range
-        function viewRangeStart () {
-            return vm.sliceFrom()+1;
-        }
-        function viewRangeEnd () {
-            return vm.sliceTo();
-        }
-
-        //Method used to display proper notification when there is no result
-        function updateResultNotification () {
-            if (vm.isUserTyping) {
-                vm.resultNotification = '';
-            }
-            else {
-                vm.resultNotification = 'There is no reference named ' + vm.query;
-            }
-        };
-
-        // Method used to prevent app from querying server with every letter input into search query panel
-        var timeout = null;
-        function timeoutRefresh()  {
-            clearTimeout(timeout);
-            vm.isUserTyping = true;
-            timeout = setTimeout(function () {
-                vm.refreshResponse();
-            }, 250);
-        };
-
-        // Method used to apply Rest response to controller variables and apply it on template
-        function refreshResponse() {
-            firstPage();
+    .controller('ReferenceSearchController', ['$scope', '$routeParams', 'openmrsRest', '$translate',
+        function ($scope, $routeParams, openmrsRest, $translate) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [This function has too many statements. (41)](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2321552673/issues/source?bid=3196969&fileBranchId=3199483#l4)